code/game/objects/items/weapons/traps.dm

```diff
@@ -689,3 +720,110 @@
 	if(!L.lying && (L.m_intent == M_RUN) || prob(5))
 		L.visible_message(SPAN_DANGER("\The [L] trips over \the [src]!"), FONT_LARGE(SPAN_DANGER("You trip over \the [src]!")))
 		L.Weaken(3)
+
+/**
+ * # Punji trap
+ *
+ * A trap that damages and gives an infection to the victim, can have a message attached
+ */
+/obj/item/trap/punji
+	name = "punji trap"
+	desc = "An horrendous trap."
+	icon = 'icons/obj/item/traps/punji.dmi'
+	icon_base = "punji"
+	icon_state = "punji0"
+	var/message = null
+
+/obj/item/trap/punji/Crossed(atom/movable/AM)
+	if(deployed && isliving(AM))
+		var/mob/living/L = AM
+		attack_mob(L)
+		update_icon()
+
+/obj/item/trap/punji/attack_mob(mob/living/L)
+
+	//Reveal the trap, if not already visible
+	hide(FALSE)
+
+	//Select a target zone
+	var/target_zone
+	if(L.lying)
+		target_zone = pick(BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG, BP_L_HAND, BP_L_ARM, BP_R_HAND, BP_R_ARM)
+	else
+		target_zone = pick(BP_L_FOOT, BP_R_FOOT, BP_L_LEG, BP_R_LEG)
+
+	//Try to apply the damage
+	var/success = FALSE
+	success += L.apply_damage(60, DAMAGE_TOXIN, target_zone, used_weapon = src, armor_pen = activated_armor_penetration)
+	success += L.apply_damage(60, DAMAGE_BRUTE, target_zone, used_weapon = src, armor_pen = activated_armor_penetration)
+
+	//If successfully applied, give the message
+	if(success)
+
+		//Show the leftover message, if any, after a little
+		addtimer(CALLBACK(src, PROC_REF(reveal_message), L), 3 SECONDS)
+
+		//Give a simple message and return if it's not a human
+		if(!ishuman(L))
+			L.visible_message(SPAN_DANGER("You step on \the [src]!"))
+			return
+
+		var/mob/living/carbon/human/human = L
+		var/obj/item/organ/organ = human.get_organ(target_zone)
+
+		human.visible_message(SPAN_DANGER("\The [human] steps on \the [src]!"),
+								SPAN_WARNING(FONT_LARGE(SPAN_DANGER("You step on \the [src], feel your body fall, and something sharp penetrate your [organ.name]!"))),
+								SPAN_WARNING("<b>You feel your body fall, and something sharp penetrate your [organ.name]!</b>"))
+
+		//If it's a human and not an IPC, apply an infection
+		//We are returning early before this step in case something isn't a human, so this should be fine not to catch borgs/bot/exosuits/whatever
+		if(!isipc(L))
+			organ.germ_level += INFECTION_LEVEL_THREE
```

Comment:
These infections are extremely powerful and completely immobilize someone within minutes. I think it's too much combined with the toxin

This IPC override should also be at the top, together with vaurca maybe. I don't think carapace or metal would be getting damaged from punji sticks

Reply:
I mean with the spear reasoning most things wouldn't damage an IPC is the issue. As for the carapace idk, I just think it's a better idea to add some species diversity to the game rather than get stuck on stuff like "is a vaurca carapace harder than steel"